site/content/entry/citra-progress-report-2020-q2.md

```diff
@@ -0,0 +1,699 @@
+[leoetlino](https://github.com/leoetlino)'s changes make it easier for modders to add custom code to 3DS games, so be sure to check out Project Restoration sometime as a thanks for these contributions!
+
+(Pic of Project restoration on Citra)
+A classic game worth another playthrough with these improvements
+
+### LayeredFS support ([#5088](https://github.com/citra-emu/citra/pull/5088)) by [zhaowenlan1779](https://github.com/zhaowenlan1779)
+
+LayeredFS is a technique used by mod makers to replace files in the RomFS (Read-only File System, where game assets and occasionally code are stored). With LayeredFS, you can replace the game graphics, music, etc without having to rebuild the entire RomFS which can be huge.
+
+Since [Luma3DS](https://github.com/AuroraWright/Luma3DS) has had LayeredFS for years, there were many requests for Citra to add it as well. That was why [zhaowenlan1779](https://github.com/zhaowenlan1779) decided to take on this task.
+
+More modern consoles often have high-level FS APIs, which means that the game requests the OS to read files. While this means more work for the initial implementation, it also means that LayeredFS can be added with ease simply by replacing the data to be returned to the game. For the 3DS, however, things are different. The FS service only provides basic functions to read the RomFS as a huge binary chunk - and the games need to parse the filesystem themselves.
+
+[Luma3DS](https://github.com/AuroraWright/Luma3DS)'s LayeredFS works by hooking the SDK code the games include to parse the RomFS, but [zhaowenlan1779](https://github.com/zhaowenlan1779) decided to take a harder and more robust approach. With help from 3dbrew documents, he wrote code to rebuild the RomFS metadata and assign fake data offsets to files. From the game's perspective, it would seem as if the RomFS has been completely rebuilt, but we are not actually doing that much work.
+
+(Picture of a game with fancy mods)
+
+## Disk Shader Cache ([#4923](https://github.com/citra-emu/citra/pull/4923)) by [jroweboy](https://github.com/jroweboy)
+
+Do you like it if you game just hanges for a few seconds repeatedly? No? Then this feature is something for you.
+
+Stuttery gameplay is a real issue in some games on Citra like Luigi's Mansion: Dark Moon.
+This game is developed with a very modern approach to GPU hardware and makes extensive use of the 3DS GPU for all of its effects.
+
+A shader is a small program that runs on your GPU. 
+Whenever the game tells Citra to run a new shader, Citra translates this shader, so that your GPU can understand it. Then, it gets uploaded to your GPU.
+This entire process can take a bit of time, so Citra has to pause emulation until the shader is ready, causing stutter whenever the game is drawing something for the first time.
```

Comment:
I'm not sure which one is better, but I've heard that `have to` more or less means that the situation caused this necessity (it's more objective), while `must` is more subjective

e.g. I had to stay at home because it was raining. vs I must work hard to earn money.